zerver/webhooks/gitlab/view.py

```diff
@@ -482,6 +511,18 @@ def get_subject_based_on_event(
             id=payload["issue"]["iid"].tame(check_int),
             title=payload["issue"]["title"].tame(check_string),
         )
+    elif event.startswith("Note Hook Design"):
+        comment = payload["object_attributes"]
+        new_path = comment["original_position"]["new_path"].value
+        assert isinstance(new_path, str)
+        new_path_split = new_path.split("/")
+        issue_number = new_path_split[1][new_path_split[1].find("-") + 1 :]
+        return TOPIC_WITH_PR_OR_ISSUE_INFO_TEMPLATE.format(
+            repo=get_repo_name(payload),
+            type="issue",
+            id=issue_number,
+            title="",
```

Comment:
As mentioned above, ideally we want the issue name to be present in the topic. But since it's not present in the payload. You should go for `project-name / design design-name`.